lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -128,6 +139,28 @@ def self.download_file_from_tag(repository:, tag:, file_path:, download_folder:)
 
         download_path
       end
+
+      # Creates (or updates an existing) GitHub PR Comment
+      def self.comment_on_pr(project_slug:, pr_number:, body:, reuse_identifier: SecureRandom.uuid)
+        client = github_client
```

Comment:
🤔 💭  The better practice in Ruby-world to do memoization and avoid the `github_client` to be reconstructed on every call would be something like this at the method definition:

```
def self.github_client
  @@client ||= begin
     … the rest of the already existing code
  end
end
```

_The `memoized_var ||= complex_value` construct has the effect of returning `memoized_var` if it already exists and is not `nil` – thus immediately returning the cached value – and if not yet cached, would evaluate the rhs `complex_value`, assign (`=`) it to the `memoized_var`, and return the value._

Usually for the `memoized_var` we use an **instance variable** (prefixed by `@`), so `@foo` or `@client` here. But since this method is a class method and this whole Helper class is not written in a way that expects us to instantiate it (as all its methods are `def self.xx` class methods), I _think_ we are expected to use `@@client` – aka a **class variable** rather than an instance variable – here? That being said I'm also remembering that class variables (`@@foo`) are not recommended in practice and might trigger a [rubocop violation](https://rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ClassVars)… so not sure what's the best way there.

---

I think the official proper solution would be to not use class methods for the Helper and instead expect us to instantiate a `GitHubHelper` instance (from the action, at call site) to be used. That would make more sense, make it easier to test as well, etc. Similar to how in Swift shared instances and singletons are usually not the best pattern and creating instances is more recommended. But then that would probably be a change outside of this PR's scope.

---

Do what you want with all that knowledge 😅 , either try the `@@client ||= …` solution I mention and see how `rubocop` reacts, or maybe SO has suggestions there as well (didn't check) for those cases… or just keep it for a future PR (though we all know how those kind of "TODOs" go…) 🤷

Reply:
Rubocop doesn't love it, but it's the cleanest option for now, so I did that in d0ed11e6fb47410dfce650c129dbe1a62e5e3517 and added an exception.

In the future it'd be nice to not have quite so many static helpers floating around, but I think doing that here would kinda clutter up this PR.